### PR TITLE
246 log date range api

### LINKED ISSUE #246: Allow logs to be retrieved using a date range
`Task Logs` or `Service Logs` currently only allows `--tail` parameter to be specified which can be used to limit the results returned. For long running services, it would be useful to also allow the user to specify a date range when requesting logs `--startTimstamp` & `--endTimestamp`

Related to #140

### PR DESCRIPTION
**Pull request by zpatrick**
**What does this pull request do?**
Adds `start` and `end` query params for service & task logs 

**How should this be tested?**
* Smoketests
* Run `l0 service logs --start 'mm/dd hh:mm' --end 'mm/dd hh:mm' api` and make sure logs returned are in the specified time frame. 

**Checklist**
- [x] Unit tests
- [x] Smoke tests (if applicable)
- ~System tests (if applicable)~
- [x] Documentation (if applicable)

closes #246

### CODE REVIEW

**Comment by zpatrick** [comment]
@sesh-kebab: changes for the year have been made

File: api/logic/task_logic_test.go [hunk 1]
```diff
@@ -196,15 +196,15 @@ func TestGetTaskLogs(t *testing.T) {
 	}
 
 	testLogic.Backend.EXPECT().
-		GetTaskLogs("e1", "t1", 100).
+		GetTaskLogs("e1", "t1", "01/01 01:01", "12/12 12:12", 100).
```
Comment (sesh-kebab) [inline api/logic/task_logic_test.go]: Should the format be updated here also to `yyyy-mm-dd`?

File: api/logic/service_logic_test.go [hunk 2]
```diff
@@ -281,15 +281,15 @@ func TestGetServiceLogs(t *testing.T) {
 	}
 
 	testLogic.Backend.EXPECT().
-		GetServiceLogs("e1", "s1", 100).
+		GetServiceLogs("e1", "s1", "01/01 01:01", "12/12 12:12", 100).
```
Comment (sesh-kebab) [inline api/logic/service_logic_test.go]: Does this need updating to `yyyy-mm-dd` format?

File: api/handlers/task_handler.go [hunk 3]
```diff
@@ -68,6 +68,8 @@ func (this *TaskHandler) Routes() *restful.WebService {
 		Doc("Return recent task logs").
 		Param(service.PathParameter("id", "identifier of the task").DataType("string")).
 		Param(service.QueryParameter("tail", "number of lines from the end to return").DataType("string")).
+		Param(service.QueryParameter("start", "The start of the time range to fetch logs (format MM/DD HH:MM)").DataType("string")).
```
Comment (sesh-kebab) [inline api/handlers/task_handler.go]: The format needs to updated here.

File: api/handlers/service_handler.go [hunk 4]
```diff
@@ -88,6 +88,8 @@ func (this *ServiceHandler) Routes() *restful.WebService {
 		Doc("Return recent service logs").
 		Param(service.PathParameter("id", "identifier of the service").DataType("string")).
 		Param(service.QueryParameter("tail", "number of lines from the end to return").DataType("string")).
+		Param(service.QueryParameter("start", "The start of the time range to fetch logs (format MM/DD HH:MM)").DataType("string")).
```
Comment (sesh-kebab) [inline api/handlers/service_handler.go]: The format needs to updated here.

**Pull request merged**